src/foam/util/SafetyUtil.java

```diff
@@ -0,0 +1,23 @@
+package foam.util;
+
+
+import java.math.BigDecimal;
+
+public class SafetyUtil{
+
+  public static int compare(Object o1 , Object o2) {
+    
+    if (o1 == null && o2 == null) {
+      return 0;
+    }
+    if (o1 instanceof Number && o2 instanceof Number) {
+      if (o1.toString() != "NaN" || o2.toString() != "NaN") {
+        return new BigDecimal(o1.toString()).compareTo(new BigDecimal(o2.toString()));
+      }
+    }
+    if (o1 instanceof String && o2 instanceof String) {
```

Comment:
This if statement doesn't do anything because if you just remove it the last statement will do the same thing anyway.